### PR TITLE
[8.x] Exponential backoffs

### PR DESCRIPTION
**Pull request by themsaid**
This PR adds the ability to set exponential backoff in on the worker command:

```
--backoff=1,5,10
```

And inside a job class:

```
$backoff = [1, 5, 10];

function backoff(){
    return [1, 5, 10];
}
```

This will instruct the worker to retry the job after 1 second on the first exception, on the second exception it'll be retried after 5 seconds, on the 3rd exception it'll be retried after 10 seconds, on any future exception it'll be retried after 10 seconds.

### CODE REVIEW

File: src/Illuminate/Queue/Worker.php [hunk 1]
```diff
@@ -497,6 +495,24 @@ protected function failJob($job, Throwable $e)
...
     }
 
+    /**
+     * Calculate the backoff for the given job.
+     *
+     * @param  \Illuminate\Contracts\Queue\Job  $job
+     * @param  \Illuminate\Queue\WorkerOptions  $options
+     * @return int
+     */
+    protected function calculateBackoff($job, WorkerOptions $options): int
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php:505]: Usually the framework doesn't use return types.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by themsaid** [comment]
I don’t use return types. It’s PHPStorm’s fault 😀

File: src/Illuminate/Queue/Worker.php [hunk 2]
```diff
@@ -497,6 +495,24 @@ protected function failJob($job, Throwable $e)
...
+     */
+    protected function calculateBackoff($job, WorkerOptions $options)
+    {
+        $backoff = explode(',',
+            method_exists($job, 'backoff') && ! is_null($job->backoff())
+                        ? $job->backoff()
+                        : $options->backoff
+        );
+
+        return $backoff[$job->attempts() - 1] ?? last($backoff);
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php:513]: isn't this returning a string?

Comment (X-Coder264) [reply src/Illuminate/Queue/Worker.php:513]: The int return type here would actually make this function return an integer as it should be returned, but no, you had to remove it 😛

**Pull request merged**